game-core/src/main/java/games/strategy/triplea/util/MovableUnitsFilter.java

```diff
@@ -0,0 +1,205 @@
+package games.strategy.triplea.util;
+
+import com.google.common.base.Preconditions;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.MoveDescription;
+import games.strategy.engine.data.PlayerId;
+import games.strategy.engine.data.Route;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitCollection;
+import games.strategy.triplea.delegate.AbstractMoveDelegate;
+import games.strategy.triplea.delegate.AbstractMoveDelegate.MoveType;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.MoveValidator;
+import games.strategy.triplea.delegate.UndoableMove;
+import games.strategy.triplea.delegate.UnitComparator;
+import games.strategy.triplea.delegate.data.MoveValidationResult;
+import games.strategy.triplea.delegate.data.MustMoveWithDetails;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.triplea.java.collections.CollectionUtils;
+
+/**
+ * Utility class for filtering a unit list to the subset of those units that can move on a route.
+ */
+public final class MovableUnitsFilter {
+  /** The result of the filter operation. */
```

Comment:
nit, what do you think about renaming this class to `FilterOperationResult`? That way we can omit the clarifying comment.

Reply:
Isn't it better to have a comment on each class anyway? I'm OK with renaming, but I'm not sure that's a good reason not to have a comment?

Reply:
Good comments are good, particularly to explain the purpose and utility of a class. On the other hand this comment I think could be baked into the code and the code would be more self documenting. In this example, a comment describing exactly what a filter operation is perhaps could be useful. It could also be after a rename that the class is self-explanatory enough that the commentary is not needed.

Reply:
Renamed the class. Kept the comment though.